core/lib/object_store/src/retries.rs

```diff
@@ -28,6 +28,43 @@ impl Request<'_> {
         self,
         store: &impl fmt::Debug,
         max_retries: u16,
+        f: F,
+    ) -> Result<T, ObjectStoreError>
+    where
+        Fut: Future<Output = Result<T, ObjectStoreError>>,
+        F: FnMut() -> Fut,
+    {
+        self.retry_internal(max_retries, f).await
+    }
+
+    #[tracing::instrument(
+        name = "object_store::Request::retry_optional",
+        skip(f), // output request and store as a part of structured logs
+        fields(retries) // Will be recorded before returning from the function
+    )]
+    async fn retry_optional<T, Fut, F>(
```

Comment:
AFAICT, this function is unused. It's not detected as such because of the `instrument` attribute.

Reply:
Removed. ✅